npm/react/plugins/utils/get-transpile-folders.js

```diff
@@ -18,7 +18,28 @@ function getTranspileFolders (config) {
     folders.push(config.supportFolder)
   }
 
-  return folders
+  // attempt to add directories based on spec pattern
```

Comment:
I think we need to find a better way to accomplish this. For example, I changed the `specPattern` in `npm/react/examples/react-scripts/cypress.config.js` to `'**/*cy-spec.js'` and the result was `dirsFromSpecPattern = '/Users/zachjw/work/cypress/npm/react/examples/react-scripts/**/*cy-spec.j'`

I'm trying to think of a more proper way to handle this but it's a pretty difficult problem. Could we use globby on the spec pattern and extract from the returned files/dirs? That approach might fall short if they haven't added a spec yet or they add a new one that results in a new folder...

Also, there is a check on line 9 that looks like it can be removed i.e. `if (config.componentFolder)`

Reply:
Removed componentFolder https://github.com/cypress-io/cypress/pull/19319/commits/9a13720f8d4c881a8edac8408c9239a353fc8ac5

Not sure on how to handle transpile folders for a spec glob... I wish CRA didn't have this silly restriction in the first place. Your globby solution sounds interesting, but not sure if this will impact perf (adding dozens, if not hundreds of directories to the watcher).

Reply:
I wonder if we can just add the `cypress` folder and call it a day.